### ISSUE TITLE
It's possible to create fields with invalid names

### ISSUE THREAD

**Issue by 527bd5** [typesense/typesense#948]
## Description

Creating a collection with a field containing `:` works but breaks certain features, at least filtering.

## Steps to reproduce

```import typesense

COLLECTION = "example"
client = typesense.Client(
    {
        "api_key": "TYPESENSEDEV",
        "nodes": [{"host": "localhost", "port": "8108", "protocol": "http"}],
        "connection_timeout_seconds": 2,
    }
)

try:
    client.collections[COLLECTION].delete()
except:
    pass

fields = [
    {
        "name": "color:foo",
        "type": "string",
    },
]
client.collections.create(
    {
        "name": COLLECTION,
        "fields": fields,
    }
)

products = [
    {"id": "1", "color:foo": "red"},
    {"id": "2", "color:foo": "blue"}
]

client.collections[COLLECTION].documents.import_(products, {"action": "create"})

hits = client.collections[COLLECTION].documents.search(
    {
        "q": "*",
        "query_by": "color:foo",
        "filter_by": "color:foo:=red"
    }
)["hits"]
```

## Expected Behavior

Filter by the field `color:foo`. Alternatively, the schema should be rejected because the field names contain unsupported characters.

## Actual Behavior

`Could not find a filter field named `color` in the schema.`

Note that trying to escape the field name doesn't seem to work either, ` is just recognized as part of the field name.

## Metadata

**Typesense Version**: 0.24.0
**OS**: Running in Docker

**Comment by mbombonato** [comment]
I was about to post something related. I'm facing the same issue when I'm trying to use 'vector_query'.

The id of my documents uses ':' and '-' as part of it. Ex:
"61940ba2e689060011f69be1:e24f244f-2bbc-46a0-9ae5-7e3ce3912106:2:61940ba2e689060011f69be1"

When I try to use "vector_query": "vector_field:([], id: 61940ba2e689060011f69be1:e24f244f-2bbc-46a0-9ae5-7e3ce3912106:2:61940ba2e689060011f69be1)" I got the following error: "Malformed vector query string."

Is there any way to fix that without having to reindex everything using another pattern for our ids? That would be a huge work from our side :(

**Comment by mbombonato** [comment]
@jasonbosco nope. Typesense probably thinks the back ticks is part of the id value.
"Document id referenced in vector query is not found."
